sktime/transformations/series/summarize.py

```diff
@@ -453,43 +453,37 @@ class description for in-depth explanation.
         if isinstance(Z, pd.core.groupby.generic.SeriesGroupBy):
             if bfill is False:
                 feat = getattr(
-                    Z.shift(lag).rolling(
-                        window=window_length, min_periods=window_length
-                    ),
+                    Z.rolling(window=window_length, min_periods=window_length),
                     summarizer,
-                )()
+                )().shift(lag)
             else:
-                feat = getattr(
-                    Z.shift(lag)
+                feat = (
+                    getattr(
+                        Z.rolling(window=window_length, min_periods=window_length),
+                        summarizer,
+                    )()
+                    .shift(lag)
                     .bfill()
-                    .rolling(window=window_length, min_periods=window_length),
-                    summarizer,
-                )()
+                )
             feat = pd.DataFrame(feat)
         else:
             if bfill is False:
                 feat = Z.apply(
                     lambda x: getattr(
-                        x.shift(lag).rolling(
-                            window=window_length, min_periods=window_length
-                        ),
+                        x.rolling(window=window_length, min_periods=window_length),
                         summarizer,
-                    )()
+                    )().shift(lag)
                 )
             else:
                 feat = Z.apply(
                     lambda x: getattr(
-                        x.shift(lag)
-                        .bfill()
-                        .rolling(window=window_length, min_periods=window_length),
+                        x.rolling(window=window_length, min_periods=window_length),
                         summarizer,
                     )()
+                    .shift(lag)
+                    .bfill()
                 )
     else:
-        if bfill is False:
-            feat = Z.shift(lag)
-        else:
-            feat = Z.shift(lag).bfill()
         if isinstance(Z, pd.core.groupby.generic.SeriesGroupBy) and callable(
             summarizer
         ):
```

Comment:
Line 490 is the cause of the third error you mentioned. In that case `feat` is not defined yet. Thus, in Line 490 and 494, it must probably be `Z.apply(...`

Reply:
Yes the `feat` not being defined yet has now been resolved I will now try to fix the other error .